File: frontend/src/lib/types/identityProof.ts
export type ProofIntensity = 'Easy' | 'Moderate' | 'Hard';

export interface IdentityProof {
	id: string;
	identityId: string;
	identityName: string;
	identityColor: string | null;
	identityIcon: string | null;
	proofDate: string;
	description: string | null;
	intensity: ProofIntensity;
	voteValue: number;
	createdAt: string;
}

export interface CreateIdentityProofRequest {
	identityId: string;
	description?: string | null;
	intensity?: ProofIntensity;
}

File: backend/src/HelpMotivateMe.Core/Entities/IdentityProof.cs
using HelpMotivateMe.Core.Enums;

namespace HelpMotivateMe.Core.Entities;

public class IdentityProof
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public Guid IdentityId { get; set; }
    public DateOnly ProofDate { get; set; }
    public string? Description { get; set; }
    public ProofIntensity Intensity { get; set; } = ProofIntensity.Easy;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public User User { get; set; } = null!;
    public Identity Identity { get; set; } = null!;
}
